src/components/data/data-utils.js

```diff
@@ -695,6 +696,82 @@ export const ModalToolButton = ({ icon, text, disabled, ...props }) => {
   ])
 }
 
+export const DeleteEntityColumnModal = ({ workspaceId: { namespace, name }, column: { entityType, attributeName }, onDismiss, onSuccess }) => {
+  const [deleting, setDeleting] = useState(false)
+  const [deleteConfirmation, setDeleteConfirmation] = useState('')
+
+  const signal = Utils.useCancellation()
+
+  const deleteColumn = async () => {
+    try {
+      setDeleting(true)
+      await Ajax(signal).Workspaces.workspace(namespace, name).deleteEntityColumn(entityType, attributeName)
+      onDismiss()
+      onSuccess()
+    } catch (e) {
+      reportError('Unable to modify column', e)
+      setDeleting(false)
+    }
+  }
+
+  return h(Modal, {
+    title: 'Delete Column',
+    onDismiss,
+    okButton: h(ButtonPrimary, {
+      onClick: deleteColumn,
+      tooltip: Utils.cond(
+        [_.toLower(deleteConfirmation) !== 'delete column', () => 'You must type the confirmation message'],
+        () => 'Delete Column')
```

Comment:
I think this might make more sense as a ternary:
```suggestion
      tooltip: _.toLower(deleteConfirmation) !== 'delete column' ? 'You must type the confirmation message' : undefined
```
In the fall-through case, I would hope you can leave it as undefined rather than duplicating the button text. Maybe the behavior bothered you, and if so, please ignore me.